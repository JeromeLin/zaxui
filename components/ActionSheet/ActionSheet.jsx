import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Popup from '../Popup';

class ActionSheet extends PureComponent {
  render() {
    const { prefixCls, className, shape, visible, onMaskClick, actions, cancelText, onCancel } = this.props;

    const cls = classnames({
      [`${prefixCls}`]: true,
      [className]: !!className,
      [`shape-${shape}`]: !!shape,
    });

    return (
      <Popup visible={visible} onMaskClick={onMaskClick}>
        <div className={cls}>
          <div className={`${prefixCls}-actions`}>
            {
              actions.map((action, index) => {
                const actionCls = classnames({
                  [`${prefixCls}-btn`]: true,
                  [`theme-${action.theme}`]: !!action.theme,
                });
                return <a key={+index} className={actionCls} onClick={action.onClick}>{action.text}</a>;
              })
            }
          </div>
          {
            typeof onCancel === 'function'
              ? (
                <div className={`${prefixCls}-cancel`}>
                  <a className={`${prefixCls}-btn`} onClick={onCancel}>{cancelText}</a>
                </div>
                )
              : null
          }
        </div>
      </Popup>
    );
  }
}

ActionSheet.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  shape: PropTypes.oneOf(['radius']),
  actions: PropTypes.arrayOf(PropTypes.object),
  cancelText: PropTypes.string,
  onCancel: PropTypes.func,
  onMaskClick: Popup.propTypes.onMaskClick,
};

ActionSheet.defaultProps = {
  prefixCls: 'za-actionsheet',
  className: null,
  shape: 'radius',
  actions: [],
  cancelText: '取消',
  onCancel: null,
  onMaskClick: Popup.defaultProps.onMaskClick,
};

export default ActionSheet;
