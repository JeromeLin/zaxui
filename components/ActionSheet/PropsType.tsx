export type Action = {
  text: string,
  theme?: 'default' | 'primary' | 'info' | 'success' | 'warning' | 'error',
  className?: string,
  onClick?: () => void,
};

export default interface PropsType {
  visible?: boolean;
  shape?: 'radius';
  actions: Action[];
  cancelText?: string;
  onMaskClick?: () => void;
  onCancel?: () => void;
}
