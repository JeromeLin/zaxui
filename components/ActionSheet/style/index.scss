@import "../../style/variables";
@import "../../style/core/index";
@import "../../Popup/style/index";
@import "./component";
