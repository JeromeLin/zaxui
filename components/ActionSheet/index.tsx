import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { ActionSheetProps } from './PropsType';
import Popup from '../Popup';

export { ActionSheetProps };

export default class ActionSheet extends PureComponent<ActionSheetProps, {}> {

  static defaultProps = {
    prefixCls: 'za-actionsheet',
    shape: 'radius',
    visible: false,
    actions: [],
    cancelText: '取消',
  }

  render() {
    const { prefixCls, className, shape, visible, onMaskClick, actions, cancelText, onCancel } = this.props;

    const cls = classnames(`${prefixCls}`, className, {
      [`shape-${shape}`]: !!shape,
    });

    return (
      <Popup visible={visible} onMaskClick={onMaskClick}>
        <div className={cls}>
          <div className={`${prefixCls}-actions`}>
            {
              actions.map((action, index) => {
                const actionCls = classnames(`${prefixCls}-btn`, {
                  [`theme-${action.theme}`]: !!action.theme,
                });
                return <a key={+index} className={actionCls} onClick={action.onClick}>{action.text}</a>;
              })
            }
          </div>
          {
            typeof onCancel === 'function' && (
              <div className={`${prefixCls}-cancel`}>
                <a className={`${prefixCls}-btn`} onClick={onCancel}>{cancelText}</a>
              </div>
            )
          }
        </div>
      </Popup>
    );
  }
}
——