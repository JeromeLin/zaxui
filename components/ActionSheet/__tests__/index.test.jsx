import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ActionSheet from '../index';

describe('ActionSheet', () => {
  it('renders correctly', () => {
    const onMaskClick = jest.fn();
    const onCancel = jest.fn();

    const wrapper = render(
      <ActionSheet
        visible
        onMaskClick={onMaskClick}
        actions={[
          {
            text: '操作一',
            onClick: () => console.log('点击操作一'),
          },
          {
            text: '操作二',
            onClick: () => console.log('点击操作二'),
          },
          {
            theme: 'error',
            text: '操作三',
            onClick: () => console.log('点击操作三'),
          },
        ]}
        onCancel={onCancel}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
