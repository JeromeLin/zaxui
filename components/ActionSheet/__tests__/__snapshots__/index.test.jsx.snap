// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet renders correctly 1`] = `
<div
  class="za-popup"
>
  <div
    class="za-popup-wrapper za-popup-wrapper-bottom"
    style="-webkit-transition-duration:200ms;transition-duration:200ms"
  >
    <div
      class="za-actionsheet"
    >
      <div
        class="za-actionsheet-actions"
      >
        <a
          class="za-actionsheet-item"
        >
          操作一
        </a>
        <a
          class="za-actionsheet-item"
        >
          操作二
        </a>
        <a
          class="za-actionsheet-item theme-error"
        >
          操作三
        </a>
      </div>
    </div>
  </div>
  <div
    class="za-mask normal"
    style="-webkit-animation-duration:200ms;animation-duration:200ms"
  />
</div>
`;
